src/embind/emval.js

```diff
@@ -65,58 +59,45 @@ var LibraryEmVal = {
     return symbol;
   },
 
-  $Emval__deps: ['$emval_freelist', '$emval_handles', '$throwBindingError', '$init_emval'],
+  $Emval__deps: ['$emval_handles', '$throwBindingError', '$init_emval'],
   $Emval: {
     toValue: (handle) => {
       if (!handle) {
           throwBindingError('Cannot use deleted val. handle = ' + handle);
       }
-  #if ASSERTIONS
-      // handle 2 is supposed to be `undefined`.
-      assert(handle === 2 || emval_handles[handle] !== undefined && handle % 2 === 0, `invalid handle: ${handle}`);
-  #endif
-      return emval_handles[handle];
+      return emval_handles.get(handle);
     },
 
     toHandle: (value) => {
       switch (value) {
-        case undefined: return 2;
-        case null: return 4;
-        case true: return 6;
-        case false: return 8;
+        case undefined: return 1;
+        case null: return 2;
+        case true: return 3;
+        case false: return 4;
         default:{
-          const handle = emval_freelist.pop() || emval_handles.length;
-          emval_handles[handle] = value;
-          emval_handles[handle + 1] = 1;
-          return handle;
+          return emval_handles.allocate(value);
         }
       }
     }
   },
 
-  _emval_incref__deps: ['$emval_handles'],
-  _emval_incref: (handle) => {
-    if (handle > {{{ EMVAL_LAST_RESERVED_HANDLE }}}) {
-      emval_handles[handle + 1] += 1;
-    }
+  _emval_clone__deps: ['$emval_handles'],
+  _emval_clone: (handle) => {
+    return Emval.toHandle(emval_handles.get(handle));
   },
```

Comment:
Make this into single line arrow function (no curly braces or return statement)

Reply:
Done